trace.js

```diff
@@ -0,0 +1,63 @@
+
+function Trace(client, name, tags, now) {
+  if (!tags) {
+    tags = [ ];
+  }
+
+  if (!now) {
+    now = new Date;
+  }
+
+  this.client = client;
+  this.name = name;
+  this.tags = tags;
+  this.start = now;
+  this.steps = [ ];
+}
+
+Trace.prototype.step = function(step, tags, now) {
+  if (!tags) {
+    tags = [ ];
+  }
+
+  if (!now) {
+    now = new Date;
+  }
+
+  tags.push('step:' + step);
+  this.steps.push({ tags: tags, time: now });
+};
+
+Trace.prototype.complete = function(now) {
+  var counter = this.name + '.count';
+  var timer = this.name + '.seconds';
+  var start = this.start;
+  var ptime = this.start;
+  var stats = this.client;
+  var steps = this.steps;
+  var tags = this.tags;
+
+  if (!stats) {
+    throw new Error("Trace.complete: called more than once");
```

Comment:
can we add the `name` to this message so it's easy to see in an error log which trace was responsible?